### PR TITLE
Support deleted users in export report

### PR DESCRIPTION
**Pull request by priyank29**
- Support deleted users in export report
- localized strings

### CODE REVIEW

File: Source/CompanyCommunicator.Prep.Func/Export/Streams/DataStreamFacade.cs [hunk 1]
```diff
@@ -66,35 +66,53 @@ public class DataStreamFacade : IDataStreamFacade
...
-            if (notificationId == null)
-            {
-                throw new ArgumentNullException(nameof(notificationId));
-            }
+            _ = notificationId ?? throw new ArgumentNullException(nameof(notificationId));
 
             var sentNotificationDataEntitiesStream = this.sentNotificationDataRepository.GetStreamsAsync(notificationId);
+            var isForbidden = false;
+
             await foreach (var sentNotifcations in sentNotificationDataEntitiesStream)
```
Comment (aosolis) [inline Source/CompanyCommunicator.Prep.Func/Export/Streams/DataStreamFacade.cs:74]: Nit: typo: sentNotification

File: Source/CompanyCommunicator.Prep.Func/Export/Streams/DataStreamFacade.cs [hunk 2]
```diff
@@ -66,35 +66,53 @@ public class DataStreamFacade : IDataStreamFacade
...
             {
-                List<User> userList = new List<User>();
+                var users = new List<User>();
+
+                // filter the recipient not found users.
+                var recipients = sentNotifcations.Where(sentNotifcation => !sentNotifcation.DeliveryStatus.Equals(SentNotificationDataEntity.RecipientNotFound, StringComparison.CurrentCultureIgnoreCase));
```
Comment (aosolis) [inline Source/CompanyCommunicator.Prep.Func/Export/Streams/DataStreamFacade.cs:79]: Is DeliveryStatus localized? Or should the comparison here be InvariantCultureIgnoreCase?

File: Source/CompanyCommunicator.Prep.Func/Export/Streams/DataStreamFacade.cs [hunk 3]
```diff
@@ -66,35 +66,53 @@ public class DataStreamFacade : IDataStreamFacade
...
+                        {
+                            users = (await this.usersService.GetBatchByUserIds(groupRecipientsByAadId))?.ToList();
+                        }
+                    }
                 }
                 catch (ServiceException serviceException)
                 {
                     if (serviceException.StatusCode != HttpStatusCode.Forbidden)
                     {
                         throw serviceException;
```
Comment (aosolis) [inline Source/CompanyCommunicator.Prep.Func/Export/Streams/DataStreamFacade.cs:101]: throw;
propagate the exception, not re-throw

File: Source/CompanyCommunicator.Prep.Func/Export/Streams/DataStreamFacade.cs [hunk 4]
```diff
@@ -121,8 +139,8 @@ await foreach (var sentNotificationDataEntities in sentNotificationDataEntitiesS
                     {
                         Id = sentNotificationDataEntity.RowKey,
                         Name = team?.Name,
-                        DeliveryStatus = this.localizer.GetString(sentNotificationDataEntity.DeliveryStatus),
-                        StatusReason = this.GetStatusReason(sentNotificationDataEntity.ErrorMessage, sentNotificationDataEntity.StatusCode.ToString()),
+                        DeliveryStatus = sentNotificationDataEntity.DeliveryStatus is null ? sentNotificationDataEntity.DeliveryStatus : this.localizer.GetString(sentNotificationDataEntity.DeliveryStatus),
```
Comment (aosolis) [inline Source/CompanyCommunicator.Prep.Func/Export/Streams/DataStreamFacade.cs:142]: Can this be `x.DeliveryStatus ?? y`?

File: Source/CompanyCommunicator.Prep.Func/Export/Streams/DataStreamFacade.cs [hunk 5]
```diff
@@ -144,44 +162,64 @@ await foreach (var sentNotificationDataEntities in sentNotificationDataEntitiesS
             var userdatalist = new List<UserData>();
             foreach (var sentNotification in sentNotificationDataEntities)
             {
-                var user = users.
-                    FirstOrDefault(user => user != null && user.Id.Equals(sentNotification.RowKey));
+                var user = users?.FirstOrDefault(user => user != null && user.Id.Equals(sentNotification.RowKey));
                 string userType = sentNotification.UserType;
 
-                // For version less than CC v4.1.2 fetch from graph or user data table.
+                // For version less than CC v4.1.2 fetch from user data table or graph.
```
Comment (aosolis) [inline Source/CompanyCommunicator.Prep.Func/Export/Streams/DataStreamFacade.cs:168]: Let's consistently capitalize Graph here

File: Source/CompanyCommunicator.Prep.Func/Export/Streams/DataStreamFacade.cs (hunk 2 above)
Comment (priyank29) [reply Source/CompanyCommunicator.Prep.Func/Export/Streams/DataStreamFacade.cs:79]: DeliveryStatus is not localized when saved in DB. It is localized only at the time it is exported.

File: Source/CompanyCommunicator.Prep.Func/Export/Streams/DataStreamFacade.cs (hunk 1 above)
Comment (priyank29) [reply Source/CompanyCommunicator.Prep.Func/Export/Streams/DataStreamFacade.cs]: done.

File: Source/CompanyCommunicator.Prep.Func/Export/Streams/DataStreamFacade.cs (hunk 3 above)
Comment (priyank29) [reply Source/CompanyCommunicator.Prep.Func/Export/Streams/DataStreamFacade.cs]: done.

File: Source/CompanyCommunicator.Prep.Func/Export/Streams/DataStreamFacade.cs (hunk 5 above)
Comment (priyank29) [reply Source/CompanyCommunicator.Prep.Func/Export/Streams/DataStreamFacade.cs]: done

File: Source/CompanyCommunicator.Prep.Func/Export/Streams/DataStreamFacade.cs (hunk 4 above)
Comment (priyank29) [reply Source/CompanyCommunicator.Prep.Func/Export/Streams/DataStreamFacade.cs:142]: As discussed, the scenario here is if null then null else localizer.

**Review by aosolis — APPROVED** [review]

**Pull request merged**